docs/src/man/basics.md

```diff
@@ -0,0 +1,2002 @@
+  "female"
+ 3
+
+julia> sdf
+6×2 SubDataFrame
+ Row │ Sex          Job
+     │ String       Int64
+─────┼────────────────────
+   1 │ male             2
+   2 │ female           2
+   3 │ male             4
+   4 │ transgender      2
+   5 │ female           2
+   6 │ female           3
+```
+
+In the above examples we have talked about *SubDataFrame*. Now in the given example we set the value 
+of 3rd-column which is `:Job` in our data frame `df1` with `[4, 5, 7, 8, 2, 1]`:
+
+```jldoctest dataframe
+julia> df1[:, 3] = [4, 5, 7, 8, 2, 1]
+6-element Vector{Int64}:
+ 4
+ 5
+ 7
+ 8
+ 2
+ 1
+
+julia> df1
+6×3 DataFrame
+ Row │ Age    Sex          Job
+     │ Int64  String       Int64
+─────┼───────────────────────────
+   1 │    80  male             4
+   2 │    98  female           5
+   3 │    78  male             7
+   4 │    95  transgender      8
+   5 │    78  female           2
+   6 │    89  female           1
+```
+
+Column `:Age` is replaced freshly allocated vector because of broadcasting assignment:
+
+```jldoctest dataframe
+julia> df1[!, :Age] .= [85, 89, 78, 58, 96, 68] # col `:Age` is replaced freshly allocated vector
+6-element Vector{Int64}:
+ 85
+ 89
+ 78
+ 58
+ 96
+ 68
+
+julia> df1
+6×3 DataFrame
+ Row │ Age    Sex          Job
+     │ Int64  String       Int64
+─────┼───────────────────────────
+   1 │    85  male             4
+   2 │    89  female           5
+   3 │    78  male             7
+   4 │    58  transgender      8
+   5 │    96  female           2
+   6 │    68  female           1
+```
+
+Note that if column name `:Customers` and `:City` is not present in `df1` then using `!` and `:` are equivalent. The 
+major difference between in-place and replace operations is that replacing columns is needed if new values have a 
+different type than the old ones. For instance here `!` works and `:` fails:
+
+```jldoctest dataframe
+julia> df1[!, :Customers] .= ["Rohit", "Akshat", "Rahul", "Aayush", "Prateek", "Anam"] # allocates a new column `:Customers` and adds it
+6-element Vector{String}:
+ "Rohit"
+ "Akshat"
+ "Rahul"
+ "Aayush"
+ "Prateek"
+ "Anam"
+
+julia> df1[:, :City] .= ["Kanpur", "Lucknow", "Bhuvneshwar", "Jaipur", "Ranchi", "Dehradoon"] # allocates the column in-place because `:City` is not present in `data`
+6-element Vector{String}:
+ "Kanpur"
+ "Lucknow"
+ "Bhuvneshwar"
+ "Jaipur"
+ "Ranchi"
+ "Dehradoon"
+
+julia> df1
+6×5 DataFrame
+ Row │ Age    Sex          Job    Customers  City
+     │ Int64  String       Int64  String     String
+─────┼───────────────────────────────────────────────────
+   1 │    85  male             4  Rohit      Kanpur
+   2 │    89  female           5  Akshat     Lucknow
+   3 │    78  male             7  Rahul      Bhuvneshwar
+   4 │    58  transgender      8  Aayush     Jaipur
+   5 │    96  female           2  Prateek    Ranchi
+   6 │    68  female           1  Anam       Dehradoon
+```
+
+Assignment of a scalar to a data frame can be done in ranges using broadcasting:
```

Comment:
please put all broadcasted assignment examples in one place.

Reply:
I am discussing everything under the broadcasting section.

Reply:
I am not sure I follow - can you please point me to the line where you have a header of the broadcasting section as I cannot find it.

Reply:
I have created in my upcoming changes.

Reply:
OK - so can you please add comments after you push the changes (as otherwise it is hard to follow). Thank you!